community-tutorials/debootstrap/01-en.md

```diff
@@ -0,0 +1,361 @@
+rm /mnt/debinst/etc/ssh/ssh_host*
+
+# Replace with the hostkeys in your system
+cp -a /etc/ssh/ssh_host* /mnt/debinst/etc/ssh/
+```
+
+## Step 2.4 - (Copy network information)
+
+Like with the SSH configuration, we update the network configuration as well.
+```bash
+cp -a /etc/network/interfaces.d/* /mnt/debinst/etc/network/interfaces.d/
+```
+Again if your configuration resides in /etc/network/interfaces, you might consider migrating.
+
+**Be aware that [Debian changed its network interface names from version 10 to 11](https://wiki.debian.org/NetworkInterfaceNames)** and therefore you might run into serious trouble. 
+
+An approach to avoid this is to use the network-manager on first boot and only after then edit the interface definition, once you know the correct naming:
+```bash
+apt install network-manager
+```
+
+## Step 2.5 - (Set hostname)
+```bash
+cp /etc/hostname /mnt/debinst/etc/hostname
+```
+Or choose a new hostname if you want to
+
+## Step 2.6 - (Mount/Copy other directories)
+
+For any other directory (e.g. /home) you might want to consider a copy or a mount inside (/etc/fstab) from a different partition.
+Also copy all required config files and data for other services that run on your machine like apache or mysql.
+
+# Step 3 - (Finalize your setup)
+
+With all config files in place it is time to finalize the installation. If you're not inside your new system chroot to it again:
+
+```bash
+chroot /mnt/debinst /bin/bash
+```
+
+The next steps are pretty much the same as for a normal Debian setup
+
+## Step 3.1 - (Set timezone)
+```bash
+dpkg-reconfigure tzdata
+```
+
+## Step 3.2 - (Set local)
+```bash
+apt install -y locales
+dpkg-reconfigure locales
+```
+
+## Step 3.3 - (Set keyboard layout)
+```bash
+apt install -y console-setup console-setup-linux
+dpkg-reconfigure locales
+```
+
+## Step 3.4 - (Set root password)
+```bash
+passwd
+```
+
+## Step 3.5 - (Install kernel and bootloader)
+
+Now let's make it bootable. This step does not (yet) modify your current setup:
+```bash
+apt install -y linux-image-amd64 firmware-linux grub2
+```
+
+With a full functioning grub lets check if it finds our new Linux. This step does also not alter your existing bootloader:
+```bash
+update-grub
+```
+
+The output should look something like:
+```
+Generating grub configuration file ...
+Found linux image: /boot/vmlinuz-5.10.0-9-amd64
+Found initrd image: /boot/initrd.img-5.10.0-9-amd64
+Found Debian 10 on /dev/sda3
+done
+```
+Carefully read if grub was able to detect your new system and your old one (in case something goes wrong).
+The example output indicates, that grub has detected our new system as well as the old system on /dev/sda3, which we can boot to in case our new system fails.
+
+If everything is correct, you can finally write your bootloader to the beginning of the disk.
+
+```bash
+# This will override your boot sector! Make sure everything is okay before you do!
+grub-install --root-directory / /dev/sdaX
```

Comment:
The output of `update-grub` looks the same, but unfortunately I get this error when running this command.

```
root@v220211184008167349:/# grub-install --root-directory / /dev/sda4
Installing for i386-pc platform.
grub-install: warning: File system `ext2' doesn't support embedding.
grub-install: warning: Embedding is not possible.  GRUB can only be installed in this setup by using blocklists.  However, blocklists are UNRELIABLE and their use is discouraged..
grub-install: error: will not proceed with blocklists.
```

Reply:
Ah of course we need to install grub to the bootsector of the disk (meaning /dev/sda instead of /dev/sda4) sorry for the confusion.
Tutorial was updated.

Reply:
Maybe I misunderstood something. I thought after the reboot I should generally be in the new Debian without `chrooted`.

Reply:
What I also noticed: the new partition is no longer mounted after the reboot.

Reply:
Yes, I assume its because you executed the `grub-install` command from your old system. Which lets your boot sector point the next stage of your booting process to your old partition. In there, the installed grub (on your /boot of your old partition) identifies your linux images of your old linux as the higher prioritised.

Since we used the `mount` command for your new partition and did not write it to fstab of your old system not having mounted it on boot is what I would expect. This is also not necessary, since we will not need the old system anyway.

So here are the steps to fix your setup:
1. Boot to your old system
2. Mount the partition of your new system
3. chroot to your new system
4. run `update-grub` and `grub-install` inside your new system. (Maybe document the output if there are further problems)
5. Reboot

Reply:
When I try `update-grub`, it gives me this:
`/usr/sbin/grub-probe: error: cannot find a device for / (is /dev mounted?).
`

Reply:
I think we're close now :)
Also mount in /dev like already once done during the tutorial. Before the system complains about other mount points, also mount them into the chroot

```bash

# Mountpoints we're gonna need from the main system (if you're currently chrooted, don't forget to use the exit command first)
 mount --make-rslave --rbind /proc /mnt/debinst/proc
 mount --make-rslave --rbind /sys /mnt/debinst/sys
 mount --make-rslave --rbind /dev /mnt/debinst/dev
 mount --make-rslave --rbind /run /mnt/debinst/run

#  After mounting don't forget to jump back into the chroot ;)
 chroot /mnt/debinst /bin/bash
 ```

Reply:
Ok. `update-grub` gives me now the same result as the one in the tutorial. 
`grub-install` instead gives me:
```
root@v220211184008167349:/# grub-install
Installing for i386-pc platform.
grub-install: error: install device isn't specified.
```

Reply:
Yeah you need to run it with the parameters of the tutorial ;)
```bash
grub-install --root-directory / /dev/sda
```

Reply:
Oh, of course. I am a little confused.
Debian 11 is now my new OS, but I cannot longer connect via ssh.

Reply:
Did you install the `ssh` package like mentioned as example under 2.1 in the tutorial? :)

You are now in a completely new system and since ssh is technically not a system default. So in theory your server could run without it. I've added a hint that installing ssh might be a good idea since in most cases this will be practical.

Reply:
Ok, I guess this was my fault. The tutorial itself worked.